### PR TITLE
feat(frontend): ConvertAmountSource component

### PR DESCRIPTION
**Pull request by DenysKarmazynDFINITY**
# Motivation

Component for handling source token amount input and all related information appearance.

Desktop:
Screenshot 2024-11-15 at 13 49 33

Mobile:
Screenshot 2024-11-15 at 13 49 41

### CODE REVIEW

File: src/frontend/src/lib/components/convert/ConvertAmountSource.svelte [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		{:else}
+			<div transition:slide={SLIDE_DURATION}>
+				{formatUSD({ value: convertAmountUSD })}
+			</div>
+		{/if}
+	</div>
+
+	<button
+		slot="balance"
+		class={`rounded px-2 py-0.5 ${isZeroBalance ? 'bg-error-subtle-alt text-error' : 'bg-zumthor text-blue-ribbon'}`}
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/components/convert/ConvertAmountSource.svelte:84]: i think that both `zumthor` and `blue-ribbon` can be replaced with the new colors

blue ribbon is text-brand-primary, and the other bg-brand-subtle (please double check)

File: src/frontend/src/lib/components/convert/ConvertAmountSource.svelte [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		on:click|preventDefault={setMax}
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/components/convert/ConvertAmountSource.svelte:85]: do we need `preventDefault`? we do the same elsewhere?

File: src/frontend/src/lib/components/convert/ConvertAmountSource.svelte (hunk 1 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/components/convert/ConvertAmountSource.svelte]: good catch, updated!

**Review by AntonioVentilii — APPROVED** [review]
LGTM, tks!

**Pull request merged**